core/src/main/java/zipkin/finagle/RawZipkinTracer.java

```diff
@@ -0,0 +1,115 @@
+import com.twitter.finagle.zipkin.core.Span;
+import com.twitter.util.Future;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TList;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.protocol.TType;
+import org.apache.thrift.transport.TMemoryBuffer;
+import scala.Function1;
+import scala.collection.Iterator;
+import scala.collection.Seq;
+import scala.runtime.AbstractFunction1;
+import scala.runtime.BoxedUnit;
+
+/**
+ * Receives the Finagle generated traces and sends them off to Zipkin.
+ *
+ * <p>Implement this by defining your class and delegating to this. For example:
+ * <pre>
+ * &#064;AutoService(com.twitter.finagle.tracing.Tracer.class)
+ * public final class HttpZipkinTracer extends SamplingTracer {
+ *
+ *  --snip--
+ *
+ *   // Default constructor for the service loader
+ *   public HttpZipkinTracer() {
+ *     this(Config.builder().build(),
+ *          DefaultStatsReceiver$.MODULE$.get().scope("zipkin.http")
+ *     );
+ *   }
+ *
+ *   HttpZipkinTracer(Config config, StatsReceiver stats) {
+ *     super(new HttpRawZipkinTracer(config.baseUrl(), stats),
+ *           config.initialSampleRate()
+ *     );
+ *   }
+ * }</pre>
+ */
+public abstract class RawZipkinTracer extends InternalZipkinTracer {
+
+  /**
+   * @param stats We generate stats to keep track of traces sent, failures and so on
+   */
+  protected RawZipkinTracer(StatsReceiver stats) {
+    super(stats);
+    this.protocolFactory = Protocols.binaryFactory(false, true, 0, stats);
+    final Counter okCounter = stats.scope("log_span").counter0("ok");
+    incrementOk = new AbstractFunction1<BoxedUnit, BoxedUnit>() {
+      @Override public BoxedUnit apply(BoxedUnit v1) {
+        okCounter.incr();
+        return v1;
+      }
+    };
+    final StatsReceiver errorReceiver = stats.scope("log_span").scope("error");
+    incrementError = new AbstractFunction1<Throwable, BoxedUnit>() {
+      @Override public BoxedUnit apply(Throwable e) {
+        errorReceiver.counter0(e.getClass().getName()).incr();
+        return BoxedUnit.UNIT;
+      }
+    };
+  }
+
+  private final TProtocolFactory protocolFactory;
+  private final Function1<Throwable, BoxedUnit> incrementError;
+  private final Function1<BoxedUnit, BoxedUnit> incrementOk;
+
+  @Override
+  public Future<BoxedUnit> sendSpans(Seq<Span> spans) {
+    try {
+      byte[] encoded = spansToThriftByteArray(spans);
+      return sendSpans(encoded)
+          .onSuccess(incrementOk)
+          .onFailure(incrementError);
+    } catch (RuntimeException | TException e) {
+      incrementError.apply(e);
+      return Future.exception(e).unit();
+    }
+  }
+
+  /**
+   * Sends a thrift encoded list of spans over the current transport.
+   */
+  protected abstract Future<BoxedUnit> sendSpans(byte[] thrift);
+
+  private byte[] spansToThriftByteArray(Seq<Span> spans) throws TException {
```

Comment:
later pull request, we can replace the base class, and by doing so use the built-in zipkin thrift (or json) encoder.